x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/expression/function/scalar/score/DecayTests.java

```diff
@@ -738,6 +765,69 @@ private static List<TestCaseSupplier> longRandomTestCases() {
         }));
     }
 
+    private static List<TestCaseSupplier> unsignedLongTestCase(
+        long value,
+        long origin,
+        long scale,
+        Long offset,
+        Double decay,
+        String functionType,
+        double expected
+    ) {
+        return List.of(
+            new TestCaseSupplier(
+                List.of(DataType.UNSIGNED_LONG, DataType.UNSIGNED_LONG, DataType.UNSIGNED_LONG, DataType.SOURCE),
+                () -> new TestCaseSupplier.TestCase(
+                    List.of(
+                        new TestCaseSupplier.TypedData(value, DataType.UNSIGNED_LONG, "value"),
+                        new TestCaseSupplier.TypedData(origin, DataType.UNSIGNED_LONG, "origin").forceLiteral(),
+                        new TestCaseSupplier.TypedData(scale, DataType.UNSIGNED_LONG, "scale").forceLiteral(),
+                        new TestCaseSupplier.TypedData(createOptionsMap(offset, decay, functionType), DataType.SOURCE, "options")
+                            .forceLiteral()
+                    ),
+                    startsWith("DecayLongEvaluator["),
+                    DataType.DOUBLE,
+                    closeTo(expected, Math.ulp(expected))
+                )
+            )
+        );
+    }
+
+    private static List<TestCaseSupplier> unsignedLongRandomTestCases() {
+        return List.of(
+            new TestCaseSupplier(List.of(DataType.UNSIGNED_LONG, DataType.UNSIGNED_LONG, DataType.UNSIGNED_LONG, DataType.SOURCE), () -> {
+                long randomValue = randomLong();
```

Comment:
let's use `randomUnsignedLongBetween()` for retrieving random unsigned longs

Reply:
Good point, using it now: https://github.com/elastic/elasticsearch/pull/134440/files#diff-7e5d41d68db3b842d13ffdb226a6c50a2f126de7017a6bdcd5e2f1b4ef1b51a0R758-R761